[Feat] Todo 설정 UI 구현

kamja0510 opened the pull request:
## Related issue 🛠
- closed #43

## Work Description ✏️
- 미완료 투두 설정 바텀시트 UI 구현
- 완료 투두 설정 바텀시트 UI 구현
- 날짜 변경 바텀시트 UI 구현
- 할일 수정 바텀시트 UI 구현

## Screenshot 📸

<table width="100%">
  <tr>
    <td>완료 바텀시트</td>
    <td>미완료 바텀시트</td>
  </tr>
  <tr>
    <td width="50%"><img alt="image" src="https://github.com/user-attachments/assets/e583a59b-9e0b-4183-bf22-1c452cfb31f3" /></td>
    <td><img alt="image" src="https://github.com/user-attachments/assets/828a7d03-75e4-4e5e-8877-0845c7bb6530" /></td>
  </tr>
  <tr>
    <td>할일 수정 바텀시트</td>
    <td>날짜 변경 바텀시트</td>
  </tr>
  <tr>
    <td><img width="756" height="1363" alt="image" src="https://github.com/user-attachments/assets/9339380a-730e-4ec1-81d8-b0757d550cf8" /></td>
    <td><video src = "https://github.com/user-attachments/assets/8d2bfa8f-e727-4eda-971a-1ab676322a8d"></td>
  </tr>
</table>

## To Reviewers 📢
api 나와서 실제 데이터 수정 로직같은 것은 api 구현하면서 하겠습니다.

kamja0510 committed:
[feat] #43 TodoSettingBottomSheet 구현

미완료/완료 투두 설정 바텀시트 UI 구현

kamja0510 committed:
[feat] #43 EditTodoBottomSheet UI 구현

kamja0510 committed:
[feat] #43 ChangeDateBottomSheet UI 구현

kamja0510 committed:
[chore] #43 ktlint

kamja0510 committed:
[chore] #43 ktlint

beom84 commented on app/src/main/java/org/android/bbangzip/presentation/type/TodoSettingActionType.kt:
```diff
@@ -0,0 +1,31 @@
+
+enum class TodoSettingActionType(
+    @DrawableRes val interactionIconResId: Int? = null,
+    @StringRes val actionName: Int? = null,
```
p5 : nullable하게 설정하신 이유가 뭔가요?

kamja0510 replied:
처음에 DIVIDER도 타입으로 만들었다가 별론거 같아서 뺐는데 그런것 처럼 나중에 둘 중 하나가 없는 목록도 있을 수도 있지 않을까 해서 그냥 나두었습니다.

beom84 commented on app/src/main/java/org/android/bbangzip/presentation/ui/todo/type/TodoSettingActionType.kt:
```diff
@@ -0,0 +1,31 @@
+import androidx.annotation.StringRes
+import org.android.bbangzip.R
+
+enum class TodoSettingActionType(
```
p5 : type class를 각 route마다 만들어서 관리하는건 어떤가요? 공통으로 사용되는 type이 아닌데 적혀있어서 찾기 불편하고 따로 분리하는 의미가 없다는 느낌이 듭니다
(의견 의견)

kamja0510 replied:
넵

beom84 reviewed (approved):
고생하셨슴다~

kamja0510 committed:
[refactor] #43 코드리뷰 반영

TodoSettingActionType.kt todo 패키지로 옮김
TodoSettingActionType의 파라미터를 non-nullable 타입으로 변경

kamja0510 committed:
[chore] #43 ktlint
